src/key.rs

```diff
@@ -0,0 +1,137 @@
+use crossterm::event::{KeyCode, KeyModifiers};
+use serde::{de, Deserialize};
+
+#[derive(Debug, Deserialize, Default)]
+#[serde(default)]
+pub struct KeyMap {
+    #[serde(deserialize_with = "deseralize_key")]
+    pub remove_previous_word: Option<Key>,
```

Comment:
These shuld be of type `Key`, not `Option<Key>` and default to the current key bindings. This would allow us to remove all the code duplication you currently have in `Test`.